insert_string_tpl.h

```diff
@@ -82,20 +82,16 @@ ZLIB_INTERNAL Pos INSERT_STRING(deflate_state *const s, const uint32_t str, uint
         val |= ((uint32_t)(strstart[2]) << 16);
         val |= ((uint32_t)(strstart[3]) << 24);
 #endif
-
         UPDATE_HASH(s, h, val);
         hm = h & s->hash_mask;
 
-        Pos head = s->head[hm];
-        if (head != idx) {
+        head = s->head[hm];
+        if (LIKELY(head != idx)) {
             s->prev[idx & s->w_mask] = head;
             s->head[hm] = idx;
-            if (strstart == strend)
-                ret = head;
-        } else if (strstart == strend) {
-            ret = idx;
```

Comment:
This was originally added to avoid problems when sliding the window, removing it is EBADIDEA.

Reply:
What part of it? And what commit are you commenting on, the first or the second?